Added installation of shared libraries

emmenlau opened the pull request:
This PR adds a minor change to the build instructions. It allows to install a shared build of OGDF into the `bin` directory. Without this PR, shared libraries are currently not installed.

Closes #57

milsen commented on cmake/coin.cmake:
```diff
@@ -56,13 +56,15 @@ endif()
 
 # installation
 set(COIN_INSTALL_LIBRARY_DIR "lib/${CMAKE_LIBRARY_ARCHITECTURE}" CACHE PATH "Installation path of COIN library")
+set(OGDF_INSTALL_BIN_DIR "bin" CACHE PATH "Installation path of OGDF library runtime")
```
```suggestion
set(COIN_INSTALL_BIN_DIR "bin" CACHE PATH "Installation path of COIN runtime targets")
```

milsen commented on cmake/coin.cmake:
```diff
@@ -56,13 +56,15 @@ endif()
+set(OGDF_INSTALL_BIN_DIR "bin" CACHE PATH "Installation path of OGDF library runtime")
 set(COIN_INSTALL_INCLUDE_DIR "include" CACHE PATH "Installation path of COIN header files (creates subdirectory)")
 set(COIN_INSTALL_CMAKE_DIR "lib/${CMAKE_LIBRARY_ARCHITECTURE}/cmake/OGDF/" CACHE PATH "Installation path of COIN files for CMake")
 mark_as_advanced(COIN_INSTALL_LIBRARY_DIR COIN_INSTALL_INCLUDE_DIR COIN_INSTALL_CMAKE_DIR)
```
```suggestion
mark_as_advanced(COIN_INSTALL_LIBRARY_DIR COIN_INSTALL_BIN_DIR COIN_INSTALL_INCLUDE_DIR COIN_INSTALL_CMAKE_DIR)
```

milsen commented on cmake/coin.cmake:
```diff
@@ -56,13 +56,15 @@ endif()
   EXPORT CoinTargets
   LIBRARY DESTINATION "${COIN_INSTALL_LIBRARY_DIR}"
   ARCHIVE DESTINATION "${COIN_INSTALL_LIBRARY_DIR}"
+  RUNTIME DESTINATION "${OGDF_INSTALL_BIN_DIR}"
```
```suggestion
  RUNTIME DESTINATION "${COIN_INSTALL_BIN_DIR}"
```

milsen commented on cmake/ogdf.cmake:
```diff
@@ -168,6 +168,7 @@ endif()
 
 # installation
 set(OGDF_INSTALL_LIBRARY_DIR "lib/${CMAKE_LIBRARY_ARCHITECTURE}" CACHE PATH "Installation path of OGDF library")
+set(OGDF_INSTALL_BIN_DIR "bin" CACHE PATH "Installation path of OGDF library runtime")
```
```suggestion
set(OGDF_INSTALL_BIN_DIR "bin" CACHE PATH "Installation path of OGDF runtime targets")
```

milsen commented on cmake/ogdf.cmake:
```diff
@@ -168,6 +168,7 @@ endif()
+set(OGDF_INSTALL_BIN_DIR "bin" CACHE PATH "Installation path of OGDF library runtime")
 set(OGDF_INSTALL_INCLUDE_DIR "include" CACHE PATH "Installation path of OGDF header files (creates subdirectory)")
 set(OGDF_INSTALL_CMAKE_DIR "lib/${CMAKE_LIBRARY_ARCHITECTURE}/cmake/OGDF/" CACHE PATH "Installation path of OGDF files for CMake")
 mark_as_advanced(OGDF_INSTALL_LIBRARY_DIR OGDF_INSTALL_INCLUDE_DIR OGDF_INSTALL_CMAKE_DIR)
```
```suggestion
mark_as_advanced(OGDF_INSTALL_LIBRARY_DIR OGDF_INSTALL_BIN_DIR OGDF_INSTALL_INCLUDE_DIR OGDF_INSTALL_CMAKE_DIR)
```

milsen reviewed (changes requested):
Thank you for your figuring this out and fixing it!
It would be nice if you could add my requested changes to your commit; other than that this looks good.

emmenlau commented:
Dear @milsen thanks for the thumbs up!

I've implemented your requested changes and squashed the commits (this is usually requested, but you did not explicitly say so, so I hope that was your expectation?).

Thanks for considering this PR, please review and/or merge!

milsen commented:
Thank you (also for squashing the commits). This PR looks good now, I'll merge it soon.
